nemo_skills/inference/server/model.py

```diff
@@ -493,10 +493,200 @@ def _send_request(
+
+    def __init__(
+        self,
+        host='127.0.0.1',
+        port='5000',
+        sandbox=None,
+        ssh_server=None,
+        ssh_key_path=None,
+        max_code_output_characters=1000,
+        code_execution_timeout=10.0,
+        max_code_executions=3,
+        stop_on_code_error=True,
+        handle_code_execution=True,
+        error_recovery=None,
+    ):
+        super().__init__(
+            host=host,
+            port=port,
+            sandbox=sandbox,
+            ssh_server=ssh_server,
+            ssh_key_path=ssh_key_path,
+            max_code_output_characters=max_code_output_characters,
+            code_execution_timeout=code_execution_timeout,
+            max_code_executions=max_code_executions,
+            stop_on_code_error=stop_on_code_error,
+            handle_code_execution=handle_code_execution,
+            error_recovery=error_recovery,
+        )
+
+        self.server_type = "openai"
+        self.oai_client = None
+        self.oai_client = self.prepare_openai(self.server_host, self.server_port)  # type: openai.OpenAI
+
+        self.model_name_server = self.get_model_name_from_server()
+        self.model = self.model_name_server
+
+        print(f"Model hosted by {self.server_type} server:", self.model)
+
+    def _single_call(
+        self,
+        prompts,
+        tokens_to_generate,
+        temperature,
+        top_p,
+        top_k,
+        repetition_penalty,
+        random_seed,
+        stop_phrases: List[str],
+    ):
+        request = {
+            'prompt': prompts,
+            'max_tokens': tokens_to_generate,
+            'temperature': temperature,
+            'top_p': top_p,
+            'top_k': top_k,
+            'num_generations': 1,  # VLLM provides 1 generation per prompt, duplicate prompts if you want more
+            'stop': stop_phrases,
+            'echo': False,
+            'repetition_penalty': repetition_penalty,
+            'frequency_penalty': 0.0,
+            'presence_penalty': 0.0,
+            'logprobs': None,
+            'logit_bias': None,
+            'seed': random_seed,
+        }
+
+        return self._send_request(request)
+
+    def _send_request(self, request):
+        if self.ssh_server and self.ssh_key_path:
+            raise NotImplementedError("SSH tunnelling is not implemented for vLLM OpenAI server.")
+
+        # temperature of 0 means greedy, but it's not always supported by the server
+        # so setting explicit greedy parameters instead
+        if request["temperature"] == 0:
+            request["temperature"] = 1.0
+            request["top_k"] = 1
+            request["top_p"] = 1.0
+
+        outputs = self.prompt_api(
+            prompt=request['prompt'],
+            max_tokens=request['max_tokens'],
+            temperature=request['temperature'],
+            top_p=request['top_p'],
+            top_k=request['top_k'],
+            num_generations=request.get('num_generations', 1),
```

Comment:
I guess defaults are not necessary since you explicitly specify everything above? So you can also make it simpler by just using something like

```
outputs = self.prompt_api(**request, parse_response=True)
```